tornado/httputil.py

```diff
@@ -243,6 +244,86 @@ def __deepcopy__(self, memo_dict):
         return self.copy()
 
 
+_cookie_value_decode = Cookie.SimpleCookie().value_decode
+_cookie_attr_reserved = Cookie.Morsel._reserved
+_cookie_attr_flags = set(["secure", "httponly"])
+
+# part of Cookie library module
+# https://docs.python.org/2.7/library/cookie.html
```

Comment:
Yes, a lot of changes for simple check, but i not found better solution. For example django project use ugly method to fix this https://github.com/django/django/blob/master/django/http/cookie.py#L76 or based on it https://github.com/hhru/tornado/blob/master/tornado/httputil.py#L68
Werkzeug uses its implementations of parse cookies https://github.com/mitsuhiko/werkzeug/blob/master/werkzeug/_internal.py#L271
Finally there is a third-party library https://github.com/sashahart/cookies
Maybe it's better solution than mine, maybe not, but many developers faced with this.

> Why does it matter how common invalid cookies are "in the real world" when you only have to deal with cookies on your own domain (i.e. ones you set)?

Because my project working on subdomain and many other projects works on other subdomains too, but all of them can set cookie to global domain (i.e. .foo.com) and this cookie may be invalid to my project. Without this changes i lost all users cookie if even one cookie is bad, and it makes me sad.

I understand your risk. I understand it you reject my pull request, but the problem should be solved and need to learn to ignore bad cookies instead raise exception and ignore all of cookies.